indexeddb-files/tabs/indexeddb-filehandle.js

```diff
@@ -0,0 +1,187 @@
+// Log utilities.
+const dbLogsEl = document.querySelector("#indexed-logs");
+
+document.querySelector("#clear-logs").onclick = () => {
+  dbLogsEl.innerText = "";
+};
+
+const dbLog = (...args) => {
+  dbLogsEl.innerText += '\n' + args.join(" ");
+};
+
+
+// IndexedDB initializations.
+var db;
+const dbReq = indexedDB.open("tempFilesDB", 3);
+
+dbReq.onerror = evt => {
+  dbLog(`ERROR: Fail to open indexedDB 'tempFilesDB' db: ${evt.target.error.message}`);
+};
+
+dbReq.onupgradeneeded = () => {
+  const db = dbReq.result;
+
+  dbLog(`Upgrade tempFilesDB.`);
+
+  if (!db.objectStoreNames.contains("tempFiles")) {
+    db.createObjectStore("tempFiles");
+  }
+};
+
+dbReq.onsuccess = () => {
+  db = dbReq.result;
+
+  dbLog(`Opened tempFiles db.`);
+
+  db.onerror = evt => {
+    dbLog(`Error during indexedDB operation: ${evt.target.error.message}`);
+  };
+};
+
+
+// Keep a reference to the last created temp file,
+// the file will be garbage collected once it is not
+// referenced anymore.
+var tempFile;
+
+function assertDefined(val, msg) {
+  if (!val) {
+    dbLog(msg);
+    throw new Error(msg);
+  }
+}
+
+function assertIndexedDBReady() {
+  assertDefined(db, `The expected IndexedDB is not currently opened`);
+}
+
+function assertOpenedFileReady() {
+  assertDefined(tempFile, `The expected tempFile object is not currently opened`);
+}
+
+// Handle the create file button.
+document.querySelector("#create-file").addEventListener("click", () => {
+  assertIndexedDBReady();
+
+  // Create a new mutable file.
+  const tempFileReq = db.createMutableFile("newTempFile", "binary");
+
+  tempFileReq.onsuccess = () => {
+    dbLog(`a new File object has been created`);
+
+    // Persist the opened file into a global var, so that the file handle will not
+    // be invalidated once this function returns.
+    tempFile = tempFileReq.result;
+  };
+});
+
+// Handle the save-data, get-metadata and persist-file buttons.
```

Comment:
This line (IIUC) refers to the following addEventListener calls. It would be better to split it so each one gets a comment.

It could be a bit more extensive too: "Handle the "save-data" button: fetch the contents of the "file-data" textarea and write it to the currently open temporary file."